ceph/rados/core_workflows.py

```diff
@@ -2195,3 +2195,135 @@ def create_inconsistent_object(self, pool_name, object_name):
+        init_crash_reports, err = daemon_host.exec_command(cmd=_cmd, sudo=True)
+        # capture initial number of crash list from crash ls-new
+        init_crash_ls = len(self.run_ceph_command("ceph crash ls-new"))
+
+        # generate crash using ceph daemon command
+        asok_cmd = (
+            f"cephadm shell -- ceph daemon /var/run/ceph/ceph-{daemon}.{id}.asok assert"
+        )
+        try:
+            out, err = daemon_host.exec_command(cmd=asok_cmd, sudo=True)
+        except Exception as err_exec:
+            if "exception" not in str(err_exec):
+                log.error(f"STDERR: {err_exec}")
+                log.error("Could not crash the daemon with ceph daemon utility")
+                return False
+
+        crash_report_post, err = daemon_host.exec_command(cmd=_cmd, sudo=True)
+        if not crash_report_post > init_crash_reports:
+            log.error("New crash report directory not generated in /var/lib/ceph/crash")
+            return False
+        log.info("New crash report generated successfully")
+
+        # get the crash report directory
+        _cmd = "cephadm shell -- ls -rt /var/lib/ceph/crash/ | tail -1"
+        dir, err = daemon_host.exec_command(cmd=_cmd, sudo=True)
+        dir = str(dir).strip()
+
+        # copy the crash directory to tmp
+        fsid = self.run_ceph_command(cmd="ceph fsid")["fsid"]
+        _cmd = f"cp -r /var/lib/ceph/{fsid}/crash/'{dir}' /tmp/"
+        daemon_host.exec_command(cmd=_cmd, sudo=True)
+
+        if manual_inject:
+            # inject the crash into the cluster manually
+            self.configure_host_as_client(daemon_host)
+            inject_cmd = f"ceph crash post -i /tmp/{dir}/meta"
+            daemon_host.exec_command(sudo=True, cmd=inject_cmd)
+            if not warning_check(timeout=100):
+                log.error("Crash warning not found even after manual injection")
+                return False
+        else:
+            # wait for 500 secs for crash to be reported in ceph health
+            if not warning_check(timeout=500):
+                log.error("Crash warning was not generated automatically")
+
+        # ensure crash detail is populated in crash ls-new
+        crash_ls_post = self.run_ceph_command("ceph crash ls-new")
+        if not len(crash_ls_post) > init_crash_ls:
+            log.error("New crash not listed in crash ls-new output")
+            return False
+
+        log.info("Daemon crash warning found in ceph health")
+        return True
+
+    def configure_host_as_client(self, host_node):
+        """
+        Purpose of this module is to configure the ceph keyring and conf
+        on a cluster host which is not installer to run ceph commands from
+        it
+        Args:
+            host_node: node object of the host which needs to be converted
+        Returns: None
+        """
+        # copy /etc/ceph/ files to input host
+        ceph_conf, _ = self.node.shell(["cat /etc/ceph/ceph.conf"])
```

Comment:
I believe that this can also be achieved by applying `_admin` label on the host.

ref : https://docs.ceph.com/en/latest/cephadm/host-management/#special-host-labels

See if you want to keep the implementation same or want to achieve the same via host labels.

Reply:
You would still need to install the ceph-common package though.

Reply:
ceph-common packages are being installed when if not already present.
If you recall, we had tried this exercise of adding the _admin label and the keyrings were not copied immediately.
I do not want to extend the duration of test by waiting for the keyrings to be copied, so I just copied them manually

Reply:
Sure. Should be fine.